frontend/tests/features/wheresMyGene/landingPage.test.ts

```diff
@@ -0,0 +1,137 @@
+
+function goToWMG(page: Page) {
+  return Promise.all([
+    page.waitForResponse(
+      (resp: { url: () => string | string[]; status: () => number }) =>
+        resp.url().includes("/wmg/v1/filters") && resp.status() === 200
+    ),
+    page.goto(`${TEST_URL}${ROUTES.WHERE_IS_MY_GENE}`),
+  ]);
+}
+test.describe("Tests for Gene Expression page", () => {
+  test("Should verify main panel components", async ({ page }) => {
+    await goToWMG(page);
+    // +Tissue button
+    await expect(page.getByTestId(ADD_GENE_BTN)).toBeVisible();
+    // +Gene button
+    await expect(page.getByTestId(ADD_GENE_BTN)).toBeVisible();
+    // survey alert
+    await expect(page.getByTestId("survey-alert-id")).toContainText(ALERT);
+    await expect(page.getByText("quick survey")).toHaveAttribute(
+      "href",
+      SURVEY_LINK
+    );
+    // default organism filter
+    await expect(
+      page.getByTestId("add-organism").locator("span")
+    ).toContainText("Homo sapiens");
+
+    // STEP 1 column
+    await expect(page.getByTestId("column-one")).toBeVisible();
+    await expect(page.getByTestId(ADD_TISSUE_LBL)).toContainText("STEP 1");
+    await expect(page.getByTestId(ADD_TISSUE_LBL)).toContainText("Add Tissues");
+
+    // STEP 2 column
+    await expect(page.getByTestId("column-two")).toBeVisible();
+    await expect(page.getByTestId(ADD_GENE_LBL)).toContainText("STEP 2");
+    await expect(page.getByTestId(ADD_GENE_LBL)).toContainText("Add Genes");
+
+    // STEP 3 column
+    await expect(page.getByTestId(EXPLORE_GENE_EXPRESSION)).toContainText(
+      "STEP 3"
+    );
+    await expect(page.getByTestId(EXPLORE_GENE_EXPRESSION)).toContainText(
+      "Explore Gene Expression"
+    );
+
+    // Download
+    await expect(page.getByTestId(LEGEND_WRAPPER)).toContainText("Download");
+    await expect(page.getByTestId("download-button")).toBeDisabled();
+
+    // Share
+    await expect(page.getByTestId(LEGEND_WRAPPER)).toContainText("Share");
+    await expect(page.getByTestId("share-button")).toBeDisabled();
+
+    // Source data
+    await expect(page.getByTestId(LEGEND_WRAPPER)).toContainText("Source Data");
+    await expect(
+      page.getByTestId("source-data-button").locator("svg")
+    ).toBeEnabled();
+
+    // Gene expression
+    await expect(page.getByTestId(LEGEND_WRAPPER)).toContainText(
+      "Gene Expression"
+    );
+    await expect(
+      page.locator('[id="visualization-color-scale"]')
+    ).toBeVisible();
+
+    // Gene expression in cells
+    await expect(page.getByTestId(LEGEND_WRAPPER)).toContainText(
+      "Expressed in Cells (%)"
+    );
+    await expect(page.locator('[id="expressed-in-cells-dots"]')).toBeVisible();
+    const actualSizes = [];
+    for (let i = 0; i < 5; i++) {
+      actualSizes.push(
+        Number(
+          await page
+            .locator('[id="expressed-in-cells-dots"]')
+            .locator("span")
+            .nth(i)
+            .getAttribute("size")
+        )
+      );
```

Comment:
we can just assert the array directly I remember, instead of using a loop 😄

Reply:
@tihuan the problem here is that .getAttribute("size") can only be applied to one element. Your suggestion will work with things textContext which can be applied to multiple elements. 
Pls let me know if you have better suggestion

Reply:
anyway gets landing page PR overline as this is handled there; I just merged to this branch because it is a continuation